src/voice/connection.rs

```diff
@@ -71,32 +76,43 @@ impl Connection {
         let url = generate_url(&mut info.endpoint)?;
 
         let mut client = ClientBuilder::from_url(&url).connect_secure(None)?;
+        let mut hello = None;
+        let mut ready = None;
         client.send_json(&payload::build_identify(&info))?;
 
-        let hello = loop {
+        loop {
+            if hello.is_some() && ready.is_some() {
```

Comment:
I'd prefer to have this check separately under the `VoiceEvent::Ready` and `VoiceEvent::Hello` match arms to prevent potentially unnecessary one or two conditional circuits on each loop iteration.